transaction-view/src/transaction_view.rs

```diff
@@ -0,0 +1,185 @@
+    }
+
+    /// Return the version of the transaction.
+    pub fn version(&self) -> TransactionVersion {
+        self.meta.version()
+    }
+
+    /// Return the number of required signatures in the transaction.
+    pub fn num_required_signatures(&self) -> u8 {
+        self.meta.num_required_signatures()
+    }
+
+    /// Return the number of readonly signed accounts in the transaction.
+    pub fn num_readonly_signed_accounts(&self) -> u8 {
+        self.meta.num_readonly_signed_accounts()
+    }
+
+    /// Return the number of readonly unsigned accounts in the transaction.
+    pub fn num_readonly_unsigned_accounts(&self) -> u8 {
+        self.meta.num_readonly_unsigned_accounts()
+    }
+
+    /// Return the number of static account keys in the transaction.
+    pub fn num_static_account_keys(&self) -> u8 {
+        self.meta.num_static_account_keys()
+    }
+
+    /// Return the number of instructions in the transaction.
+    pub fn num_instructions(&self) -> u16 {
+        self.meta.num_instructions()
+    }
+
+    /// Return the number of address table lookups in the transaction.
+    pub fn num_address_table_lookups(&self) -> u8 {
+        self.meta.num_address_table_lookups()
+    }
+
+    /// Return the slice of signatures in the transaction.
+    pub fn signatures(&self) -> &[Signature] {
+        let data = self.data();
+        // SAFETY: `meta` was created from `data`.
+        unsafe { self.meta.signatures(data) }
+    }
+
+    /// Return the slice of static account keys in the transaction.
+    pub fn static_account_keys(&self) -> &[Pubkey] {
+        let data = self.data();
+        // SAFETY: `meta` was created from `data`.
+        unsafe { self.meta.static_account_keys(data) }
+    }
+
+    /// Return the recent blockhash in the transaction.
+    pub fn recent_blockhash(&self) -> &Hash {
+        let data = self.data();
+        // SAFETY: `meta` was created from `data`.
+        unsafe { self.meta.recent_blockhash(data) }
+    }
+
+    /// Return an iterator over the instructions in the transaction.
+    pub fn instructions_iter(&self) -> InstructionsIterator {
+        let data = self.data();
+        // SAFETY: `meta` was created from `data`.
+        unsafe { self.meta.instructions_iter(data) }
+    }
+
+    /// Return an iterator over the address table lookups in the transaction.
+    pub fn address_table_lookup_iter(&self) -> AddressTableLookupIterator {
+        let data = self.data();
+        // SAFETY: `meta` was created from `data`.
+        unsafe { self.meta.address_table_lookup_iter(data) }
+    }
+
+    /// Return the full serialized transaction data.
+    #[inline]
+    pub fn data(&self) -> &[u8] {
+        self.data.data()
+    }
+
+    /// Return the serialized **message** data.
+    /// This does not include the signatures.
+    pub fn message_data(&self) -> &[u8] {
+        &self.data()[usize::from(self.meta.message_header.offset)..]
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
kept tests relatively simple since this is just a wrapper.
the `TransactionMeta` tests do more thorough testing of the parsing.